stream-chat-android-state/src/test/java/io/getstream/chat/android/offline/plugin/logic/channel/internal/ChannelStateLogicImplTest.kt

```diff
@@ -24,31 +24,40 @@ import io.getstream.chat.android.client.models.User
 import io.getstream.chat.android.client.setup.state.ClientState
 import io.getstream.chat.android.client.test.randomChannel
 import io.getstream.chat.android.client.test.randomMessage
+import io.getstream.chat.android.client.test.randomTypingStartEvent
 import io.getstream.chat.android.client.test.randomUser
 import io.getstream.chat.android.offline.message.attachments.internal.AttachmentUrlValidator
 import io.getstream.chat.android.offline.model.channel.ChannelData
 import io.getstream.chat.android.offline.plugin.state.channel.internal.ChannelMutableState
 import io.getstream.chat.android.offline.plugin.state.global.internal.GlobalMutableState
+import io.getstream.chat.android.test.TestCoroutineRule
 import io.getstream.chat.android.test.randomCID
 import io.getstream.chat.android.test.randomDate
 import io.getstream.chat.android.test.randomDateAfter
 import io.getstream.chat.android.test.randomDateBefore
 import io.getstream.chat.android.test.randomInt
+import kotlinx.coroutines.ExperimentalCoroutinesApi
 import kotlinx.coroutines.flow.MutableStateFlow
 import org.amshove.kluent.`should be equal to`
 import org.amshove.kluent.`should not be equal to`
+import org.junit.Rule
 import org.junit.jupiter.api.BeforeEach
 import org.junit.jupiter.api.Test
 import org.mockito.kotlin.any
 import org.mockito.kotlin.doAnswer
 import org.mockito.kotlin.doReturn
 import org.mockito.kotlin.mock
+import org.mockito.kotlin.times
 import org.mockito.kotlin.verify
 import org.mockito.kotlin.whenever
 import java.util.Date
 
+@OptIn(ExperimentalCoroutinesApi::class)
 internal class ChannelStateLogicImplTest {
 
+    @Rule
+    val testCoroutines: TestCoroutineRule = TestCoroutineRule()
+
```

Comment:
We are mixing jUnit4 and jUnit5.
On this test-sWe are mixing jUnit4 and jUnit5.
On this test-suite we are using `org.junit.jupiter.api.Test` (jUnit5) and adding a `org.junit.Rule` (jUnit4).
Instead of that, we should use a `org.junit.jupiter.api.extension.ExtensionContext` (jUnit5) and in our code we have this one `TestCoroutineExtension`uite we are using `org.junit.jupiter.api.Test` (jUnit5) and adding a `org.junit.Rule` (jUnit4).
Instead of that, we should use a `org.junit.jupiter.api.extension.ExtensionContext` (jUnit5) and in our code we have this one `TestCoroutineExtension`

Reply:
Switched to Junit5 extensions